cli/command_snapshot_copy.go

```diff
@@ -0,0 +1,190 @@
+package cli
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/snapshot"
+)
+
+var (
+	snapshotCopyCommand     = snapshotCommands.Command("copy", "Copy snapshot history from another user or host").Alias("cp")
+	snapshotCopyDryRun      = snapshotCopyCommand.Flag("dry-run", "Do not actually copy snapshots, only print what would happen").Short('n').Bool()
+	snapshotCopyMove        = snapshotCopyCommand.Flag("move", "Delete original copied snapshots").Bool()
+	snapshotCopySource      = snapshotCopyCommand.Arg("source", "Source (user@host or user@host:path)").Required().String()
+	snapshotCopyDestination = snapshotCopyCommand.Arg("destination", "Destination (defaults to current user@host)").String()
+)
+
+// runSnapshotCopyCommand copies snapshot manifests of the specified source
+// to the respective destination. This is typically used when renaming a host,
+// switching username or moving directory around to maintain snapshot history.
+//
+// Both source and destination can be specified using user@host, @host or user@host:/path
+// where destination values override the corresponding parts of the source, so both targeted
+// and mass copying is supported.
+//
+// Source:             Destination         Behavior
+// ---------------------------------------------------
+// @host1              @host2              copy snapshots from all users of host1
+// @host1              user2@host2         (disallowed as it would potentially collapse users)
+// @host1              user2@host2:/path2  (disallowed as it would potentially collapse paths)
+//
+// user1@host1         @host2              copy all snapshots to user1@host2
+// user1@host1         user2@host2         copy all snapshots to user2@host2
+// user1@host1         user2@host2:/path2  (disallowed as it would potentially collapse paths)
+//
+// user1@host1:/path1  @host2              copy to user1@host2:/path1
+// user1@host1:/path1  user2@host2         copy to user2@host2:/path1
+// user1@host1:/path1  user2@host2:/path2  copy snapshots from single path.
```

Comment:
It'd be good to put this comment in the command documentation instead. That way it is available as part of the help for the command.

Reply:
done